### PR TITLE
[APO-1040] Prune edges when node fail during serialization

### PR DESCRIPTION
**Pull request by vincent0426**
# Fix node invalidation logic to only mark invalid when serialized_node is null

## Summary

This PR refines the node invalidation logic during workflow serialization to be more granular. Previously, nodes were marked as invalid (and their edges pruned) for ANY error during serialization. Now, nodes are only marked invalid when the entire serialization completely fails (`serialized_node` is null), not for attribute-level validation errors.

**Key Changes:**
- **Modified error handling**: Separated attribute-level errors from complete node failures
- **Updated invalidation logic**: Only mark nodes invalid when `NotImplementedError` or `NodeValidationError` are raised during `serialize()`
- **Fixed test expectations**: Updated test to expect edges to remain when only attribute errors occur
- **Added comprehensive test**: New test verifies edges are pruned only for complete node failures

This addresses feedback from @dvargas92495 that edges shouldn't be removed for attribute-level errors like missing files, only when the entire node is lost.

## Review & Testing Checklist for Human

- [ ] **Verify behavioral change is correct**: Manually test that nodes with attribute errors (like missing files) retain their edges while nodes with complete serialization failures have edges pruned
- [ ] **Test existing workflows**: Ensure existing workflows that may have relied on the old invalidation behavior still work correctly  
- [ ] **Validate new test scenario**: Confirm the new test `test_serialize_node__with_complete_node_failure_prunes_edges` actually tests complete node serialization failure (not just attribute errors)
- [ ] **Edge case testing**: Test various error scenarios to ensure the distinction between attribute errors and complete failures works as expected

---

### Diagram

```mermaid
%%{ init : { "theme" : "default" }}%%
graph TD
    A["base_workflow_display.py<br/>serialize() method"]:::major-edit
    B["types.py<br/>add_error() method"]:::major-edit  
    C["code_execution_node.py<br/>error handling"]:::minor-edit
    D["test_code_execution_node.py<br/>test expectations"]:::major-edit
    E["Node Serialization Flow"]:::context
    
    A -->|"calls serialize() on nodes"| E
    E -->|"attribute error"| B
    E -->|"complete failure"| A
    A -->|"marks node invalid only<br/>for complete failures"| F["Invalid Node Tracking"]:::context
    B -->|"logs error without<br/>marking node invalid"| G["Error Collection"]:::context
    C -->|"reports missing file<br/>as attribute error"| B
    D -->|"tests both<br/>scenarios"| H["Test Coverage"]:::context

    subgraph Legend
        L1[Major Edit]:::major-edit
        L2[Minor Edit]:::minor-edit  
        L3[Context/No Edit]:::context
    end

    classDef major-edit fill:#90EE90
    classDef minor-edit fill:#87CEEB  
    classDef context fill:#FFFFFF
```

### Notes

- This change affects the fundamental behavior of workflow serialization error handling
- The distinction between attribute-level errors and complete node failures is critical for correct edge pruning
- All tests pass including the new test that specifically validates complete node failure scenarios
- Pre-commit hooks and CI checks all passed successfully

**Session Info**: Requested by vincent@vellum.ai (@vincent0426)  
**Devin Session**: https://app.devin.ai/sessions/b7412b8379bf45eab659d4de8325584c

### CODE REVIEW

File: ee/vellum_ee/workflows/display/nodes/vellum/tests/test_code_execution_node.py [hunk 1]
```diff
@@ -253,16 +253,7 @@ class Workflow(BaseWorkflow):
...
-                {
-                    "id": "ab6ef06e-df2c-4877-9c3e-9d7261b39748",
-                    "source_node_id": "9b9e2a5d-01a4-46b2-80a3-d9484b2c0e08",
-                    "source_handle_id": "3e2a3f52-5047-4e2e-9a21-37bd43c63250",
-                    "target_node_id": "ac90c0ce-f393-438c-a24f-e5e9a9286182",
-                    "target_handle_id": "3a39ea63-9f86-4891-a902-0216a7190720",
-                    "type": "DEFAULT",
-                }
-            ],
+            "edges": [],
```
Comment (dvargasfuertes) [inline ee/vellum_ee/workflows/display/nodes/vellum/tests/test_code_execution_node.py]: [blocking] we only need to do this if we lose the node entirely. now that our `add_error` is granularly scoped to just the attribute, the edge is now valid

**Review by dvargas92495 — APPROVED** [review]
